vpr/src/route/rr_graph_reader.cpp

```diff
@@ -195,6 +216,104 @@ void load_rr_file(const t_graph_type graph_type,
     }
 }
 
+void process_nodes_and_switches_bin(FILE* fp,
+                                    int* wire_to_rr_ipin_switch,
+                                    bool is_global_graph,
+                                    const std::vector<t_segment_inf>& segment_inf,
+                                    int numSwitches) {
+    auto& device_ctx = g_vpr_ctx.mutable_device();
+    uint32_t magic_num;
+    uint16_t format_version;
+    uint16_t header_length;
+    uint64_t num_rr_nodes;
+    fread(&magic_num, sizeof(magic_num), 1, fp);
```

Comment:
One more thing from my side. I have tried to compile VPR with this patch and numerous warnings, related to the unused result of `fread`, showed up.
There are two ways to solve this:
- Disable warnings related to unused results (which I am not fund of);
- Add a wrapper helper function (to [`vpr_utils.cpp`](https://github.com/SymbiFlow/vtr-verilog-to-routing/blob/master%2Bwip/vpr/src/util/vpr_utils.cpp) for example) that checks the result of `fread` and, in case, throws an error. It would be something like this:

```
 void fread_secure(void *var, size_t size, unsigned int count, FILE *fp) {                                                                                                                                                                                                                
     auto result = fread(var, size, count, fp);                                                                                                                                                                                                                                           
     if (result != count) {                                                                                                                                                                                                                                                               
         VPR_THROW(VPR_ERROR_OTHER, "ERROR reading file\n");                                                                                                                                                                                                                                
     }                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                          
     return;                                                                                                                                                                                                                                                                              
 }  
``` 
Then change all the `fread` into `fread_secure` and this solves the warning problem and makes the solution more robust IMO.

Reply:
Done